lib/miq_expression.rb

```diff
@@ -804,17 +804,15 @@ def self.quote(val, typ)
   def self.quote_human(val, typ)
     case typ&.to_sym
     when :integer, :decimal, :fixnum, :float
-      return val.to_i unless val.to_s.number_with_method? || typ == :float
+      return val if val.kind_of?(Numeric)
+      return val.to_i unless val.to_s.number_with_method?
 
-      if val =~ /^([0-9.,]+)\.([a-z]+)$/
-        val, sfx = $1, $2
-        if sfx.ends_with?("bytes") && FORMAT_BYTE_SUFFIXES.key?(sfx.to_sym)
-          "#{val} #{FORMAT_BYTE_SUFFIXES[sfx.to_sym]}"
-        else
-          "#{val} #{sfx.titleize}"
-        end
+      String::NUMBER_WITH_METHOD_REGEX.match(val.to_s)
+      val, sfx = $1, $2
```

Comment:
If we're using `.match`, then we shouldn't use `$` variables. Prefer
```suggestion
      val, sfx = String::NUMBER_WITH_METHOD_REGEX.match(val.to_s).captures
```